frontend/packages/llama-stack-modular-ui/bff/internal/api/vector_db_handler_test.go

```diff
@@ -0,0 +1,389 @@
+	assert.NoError(t, err)
+
+	var errorResponse ErrorEnvelope
+	err = json.Unmarshal(body, &errorResponse)
+	assert.NoError(t, err)
+
+	assert.Equal(t, "500", errorResponse.Error.Code)
+	assert.Equal(t, "the server encountered a problem and could not process your request", errorResponse.Error.Message)
+}
+
+func TestRegisterVectorDBHandlerMissingVectorDBID(t *testing.T) {
+	app := createTestAppForVectorDB()
+
+	// Create request with missing vector_db_id
+	registrationRequest := VectorDBRegistrationRequest{
+		EmbeddingModel: "text-embedding-ada-002",
+		// VectorDBID missing
+	}
+
+	reqBody, err := json.Marshal(registrationRequest)
+	assert.NoError(t, err)
+
+	rr := httptest.NewRecorder()
+	req, err := http.NewRequest(http.MethodPost, "/api/v1/vector-dbs", bytes.NewBuffer(reqBody))
+	assert.NoError(t, err)
+	req.Header.Set("Content-Type", "application/json")
+
+	// Add HTTP client to context
+	httpClient := &integrations.HTTPClient{}
+	ctx := context.WithValue(req.Context(), constants.LlamaStackHttpClientKey, httpClient)
+	req = req.WithContext(ctx)
+
+	// Execute handler
+	app.RegisterVectorDBHandler(rr, req, nil)
+
+	// Should return 400 error
+	assert.Equal(t, http.StatusBadRequest, rr.Code)
+
+	// Parse error response
+	body, err := io.ReadAll(rr.Body)
+	assert.NoError(t, err)
+
+	var errorResponse ErrorEnvelope
+	err = json.Unmarshal(body, &errorResponse)
+	assert.NoError(t, err)
+
+	assert.Equal(t, "400", errorResponse.Error.Code)
+	assert.Contains(t, errorResponse.Error.Message, "vector_db_id is required")
+}
+
+func TestRegisterVectorDBHandlerMissingEmbeddingModel(t *testing.T) {
+	app := createTestAppForVectorDB()
+
+	// Create request with missing embedding_model
+	registrationRequest := VectorDBRegistrationRequest{
+		VectorDBID: "test-db",
+		// EmbeddingModel missing
+	}
+
+	reqBody, err := json.Marshal(registrationRequest)
+	assert.NoError(t, err)
+
+	rr := httptest.NewRecorder()
+	req, err := http.NewRequest(http.MethodPost, "/api/v1/vector-dbs", bytes.NewBuffer(reqBody))
+	assert.NoError(t, err)
+	req.Header.Set("Content-Type", "application/json")
+
+	// Add HTTP client to context
+	httpClient := &integrations.HTTPClient{}
+	ctx := context.WithValue(req.Context(), constants.LlamaStackHttpClientKey, httpClient)
+	req = req.WithContext(ctx)
+
+	// Execute handler
+	app.RegisterVectorDBHandler(rr, req, nil)
+
+	// Should return 400 error
+	assert.Equal(t, http.StatusBadRequest, rr.Code)
+
+	// Parse error response
+	body, err := io.ReadAll(rr.Body)
+	assert.NoError(t, err)
+
+	var errorResponse ErrorEnvelope
+	err = json.Unmarshal(body, &errorResponse)
+	assert.NoError(t, err)
+
+	assert.Equal(t, "400", errorResponse.Error.Code)
+	assert.Contains(t, errorResponse.Error.Message, "embedding_model is required")
+}
+
+func TestGetAllVectorDBsHandlerSuccess(t *testing.T) {
+	app := createTestAppForVectorDB()
+
+	// Create request
+	rr := httptest.NewRecorder()
+	req, err := http.NewRequest(http.MethodGet, "/api/v1/vector-dbs", nil)
```

Comment:
can we use  apiPrefix+`VectorDBListPath` constants in here and in respective handler code too, instead of hardcoding `/api/v1/vector-dbs`, so that its easier to maintain in future wherein we need to change paths just in one place.